daffodil-test/src/test/resources/org/apache/daffodil/section12/aligned_data/Aligned_Data.tdml

```diff
@@ -267,9 +267,9 @@
     <xs:element name="e16">
       <xs:complexType>
         <xs:sequence>
-          <xs:element name="e14" type="xs:string" dfdl:terminator=":" dfdl:representation="text" dfdl:lengthKind="explicit" dfdl:length="5" dfdl:encoding="ascii" 
```

Comment:
Ok, so looking in general at these test changes, you are not distinguishing redundant properties from irrelevant properties. E.g., having dfdl:representation="text" on a string is redundant. Having "binaryBooleanTrueRep" on a string is irrelevant, and indicative of almost certainly an error. (In fact a switch to escalate such from warning to error might be of value.)

One could also warn about properties that are unnecessary because they match what is found in the default format surrounding the schema component, and so the properties do not need to be expressed on the schema component. From what I see in this code you are not going that far. I'm fine with that. 

So the question here I guess is "Am I understanding this correctly?"

Reply:
I don't do any checks for redundant properties, though that's an interesting idea. However, there are probably cases were someone wants to be explicit about a certain property on an element even though it's redundant to the default properties. We could add it, but i'm not sure it gives as much benefit as this check.

dfdl:representation="text" was removed anytime it was on a type with xs:string or xs:hexBinary. Elements with these types have an implied representation (text or binary respectively) and so we never access the representation property. So this is correct as far as warning about an unused property. This particular one probably never means a user made an error though.